lang/src/splgen/generator.py

```diff
@@ -0,0 +1,241 @@
+from dataclasses import dataclass
+from datetime import datetime
+import json
+from typing import Dict, List, Set
+from constants import *
+from spl import Script, AddressSet, Percentage, Address
+import re
+
+package_cache = {}
+
+
+def generate(model: Script, output_file: str):
+    dependencies: Set[Dependency] = set()
+    manager = model.administration.main_administrator
+    contracts = {
+        c.name: Contract(
+            name=c.name,
+            functions={
+                **{
+                    f.name: Function(
+                        name=f.name,
+                        params={p.name: format_param(p.literal)
+                                for p in f.params.parameters},
+                        mod=set()
+                    ) for f in c.implementation.methods
+                },
+                **{
+                    admin.method.name: Function(
+                        name=admin.method.name,
+                        params={p.name: format_param(p.literal)
+                                for p in admin.method.params.parameters},
+                        is_modifier=True,
+                        mod=set()
+                    ) for admin in model.administration.extension_administrators
+                },
+                manager.method.name: Function(
+                    name=manager.method.name,
+                    params={p.name: format_param(p.literal)
+                            for p in manager.method.params.parameters},
+                    is_modifier=True,
+                    mod=set()
+                )
+            },
+            dependencies=set(),
+        ).resolve_code_and_dependencies(dependencies)
+        .insert_params() for c in model.contract_section.contracts
+    }
+
+    for ca in manager.contract_administrators:
+        for method in ca.methods:
+            contracts[ca.contract.name].functions[method.name].mod.add(
+                manager.method.name)
+
+    for admin in model.administration.extension_administrators:
+        for ca in admin.contract_administrators:
+            for method in ca.methods:
+                contracts[ca.contract.name].functions[method.name].mod.add(
+                    admin.method.name)
+
+    for contract in contracts.values():
+        for fun in contract.functions.values():
+            fun.insert_modifiers()
+
+    with open(output_file, 'w') as f:
```

Comment:
Why didn't we generate using jinja?

Reply:
I thought Jinja would be overkill for a single template that has 5 lines.